src/core/varlink-metrics.c

```diff
@@ -20,7 +39,67 @@ int vl_method_list(sd_varlink *link, sd_json_variant *parameters, sd_varlink_met
                 link,
                 SD_JSON_BUILD_PAIR_STRING("name", "units"),
                 SD_JSON_BUILD_PAIR_UNSIGNED("value", hashmap_size(m->units)),
-                SD_JSON_BUILD_PAIR("fields",
-                        SD_JSON_BUILD_OBJECT(
-                                SD_JSON_BUILD_PAIR_STRING("state", "active"))));
+                SD_JSON_BUILD_PAIR(
+                        "fields",
+                        SD_JSON_BUILD_OBJECT(SD_JSON_BUILD_PAIR_STRING("state", "active"))),
+                SD_JSON_BUILD_PAIR_STRING("name", "units"),
+                SD_JSON_BUILD_PAIR_UNSIGNED("value", set_size(m->failed_units)),
+                SD_JSON_BUILD_PAIR(
+                        "fields",
+                        SD_JSON_BUILD_OBJECT(SD_JSON_BUILD_PAIR_STRING("state", "failed"))),
```

Comment:
active/failed are the unit state. While automount/device/mount/path/scope/service/slice/socket/etc are unit types. Please split accordingly.